profile/CONTRIBUTING.md

```diff
@@ -141,39 +144,103 @@ conventions.
+  - name branch prefix with issue identifier (```issue59-some-issue-summary```)
+  - reference the issue within the description of the pull request
+  - do as many request reviews -> get comments from reviewer -> fix and answer
     comments -> request reviews as necessary (until you get the reviewer(s)
     approval
-* update your [Github notification settings with Custom
+- update your [Github notification settings with Custom
   Routing](https://github.com/settings/notifications/custom_routing) from the
   ai-cfia org to your inspection.gc.ca email
-* single line commits preferably prefixed with issues being worked on or fixed
-  * example: `Fixes #19: Introduction Jolan Thomassin`
-* don't delete branches when done (leave them for future reference)
-* when feedback, suggestions or feature request that come up in one of your PR
+- single line commits preferably prefixed with issues being worked on or fixed
+  - example: `Fixes #19: Introduction Jolan Thomassin`
+- don't delete branches when done (leave them for future reference)
+- when feedback, suggestions or feature request that come up in one of your PR
   by a reviewer that would require a lot more work outside the scope of the
   current issue, it is encouraged to create a new issue capturing that
   non-trivial piece of work and document it as "postponed work" in the PR.
+
+### Handling divergent branches
+
+When your branch diverges from the main branch (due to updates in the main
+branch while you're working on your branch), follow these steps to rebase your
+branch onto the main branch. This keeps the project history clean and linear.
+
+```ascii
+main      A---B---C---D
+                       \
+branch             E---F---G (divergent)
+```
+
+A, B, C, D are commits on the main branch, and E, F, G are your commits on the branch. C and D are new commits added to the main branch after you created your branch.
+
+Steps to manually Rebase and Merge
+
+1. Fetch Latest main branch:
+
+```sh
+git fetch origin main
+```
+
+2. Rebase Your Branch:
+
+Start the rebase process to replay your branch's changes (E, F, G) on top of the
+current state of the main branch (D).
+
+```sh
+git rebase origin/main
+```
+
+3. Resolve Conflicts (if any):
+
+```ah
+git add <file>
+git rebase --continue
+```
+
+Repeat until all conflicts are resolved and the rebase completes.
+
+Push Changes:
+
+Since rebase rewrites history, you'll need to force push your branch.
+
+```sh
+git push --force-with-lease
+```
+
+This process ensures your changes maintains a linear history :
+
+```ascii
+main      A---B---C---D
+                                     \
+branch                            E'---F'---G' (rebased)
+```
+
+#### Configuring Git for Automatic Rebase
+
+To set up Git to automatically rebase instead of creating merge commits when doing a git pull, you can use the following global configuration command:
+
+```sh
+git config --global pull.rebase true
+```
```

Comment:
Explanation is clear and concise 👍🏻. Every steps to rebase in command line is well explain. Will there be a equivalent for interface?

For example:

git config --global pull.rebase true ; but you can also achieve with that specific `VScode GitHub extension feature`

Reply:
It would be great to have the UI equivalent, save more time.